app/base/handlers.js

```diff
@@ -19,10 +19,17 @@ exports.home = (req, res, next) => {
   Promise.all([Deployment.list(), User.get(req.user.auth0_id)])
     .then(([tools, user]) => {
       ns.run(() => {
+        const is_warehouse = bucket => !!bucket.s3bucket.is_data_warehouse;
+        const warehouse_buckets = user.users3buckets.filter(bucket => is_warehouse(bucket));
+        const webapp_buckets = user.users3buckets.filter(bucket => !is_warehouse(bucket));
```

Comment:
You could sort the buckets into two groups like this:
```javascript
const buckets = user.users3buckets.reduce(function (buckets, bucket) {
  const group = !!bucket.s3bucket.is_data_warehouse ? 'warehouse' : 'webapp';
  buckets[group] = buckets[group] || [];
  buckets[group].push(bucket);
  return buckets;
}, {});

res.render(..., {
  ...,
  buckets,
});
```

Reply:
... also known as "Andy, could you tidy this up for me please?"

😀

Thanks. 👍